src/lazy/streaming_raw_reader.rs

```diff
@@ -241,6 +247,11 @@ impl<Encoding: Decoder, Input: IonInput> StreamingRawReader<Encoding, Input> {
                 }
             }
 
+            // At this point, `self.input` is no longer being modified.
+            // We can hand out a reference to it that's good for the duration of 'top.
+            let io_buffer_handle: &dyn IoBufferHandle = self.input.get_mut();
+            unsafe { context.set_io_buffer_handle(io_buffer_handle) };
```

Comment:
🪧 When the streaming reader has finished modifying the input buffer, it gets a fat pointer to it and stores it in the `EncodingContext`. If the user asks to make a `LazyElement`, it will use that fat pointer to get a reference-counted copy of the input buffer.